// @brave/n8n-nodes-brave-search/dist/nodes/BraveSearch/operations/suggest/parameters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const data_1 = require("./data");
const parameters = [];
const optional_parameters = [];
parameters.push({
    displayName: 'Query',
    name: 'query',
    type: 'string',
    default: '',
    description: 'The user’s suggest search query term. Query can not be empty. The max query length is 400 characters, and the word limit is 50.',
    required: true,
}, {
    displayName: 'Count',
    name: 'count',
    type: 'number',
    default: 50,
    description: 'The number of suggestion search results returned in response. The actual number of results delivered may be less than requested. Minimum is 1, maximum is 20. The default is 5.',
    typeOptions: {
        minValue: 1,
        maxValue: 20,
    },
});
optional_parameters.push({
    displayName: 'Country',
    name: 'country',
    type: 'options',
    default: 'US',
    description: 'An associated country code. This is used as a hint. Default is "US".',
    options: data_1.SuggestCountryCodes.map(({ country, code }) => {
        return {
            name: `${country} (${code})`,
            value: code,
        };
    }),
}, {
    displayName: 'Language',
    name: 'lang',
    type: 'options',
    default: 'en',
    description: 'An associated 2-or-more character language code. Default is "en".',
    options: data_1.SuggestLanguageCodes.map(({ language, code }) => {
        return {
            name: `${language} (${code})`,
            value: code,
        };
    }),
}, {
    displayName: 'Rich Results',
    name: 'rich',
    type: 'boolean',
    default: false,
    description: 'Whether to enhance suggestions with rich results',
    hint: 'Requires a paid <em>Suggest</em> subscription',
});
if (optional_parameters.length > 0) {
    parameters.push({
        displayName: 'Additional Parameters',
        name: 'additionalParameters',
        type: 'collection',
        default: {},
        placeholder: 'Add Parameter',
        options: optional_parameters,
    });
}
exports.default = parameters;
