// @brave/n8n-nodes-brave-search/dist/nodes/BraveSearch/operations/spellcheck/parameters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const data_1 = require("./data");
const parameters = [];
const optional_parameters = [];
parameters.push({
    displayName: 'Query',
    name: 'query',
    type: 'string',
    default: '',
    description: 'The user’s spellcheck search query. Query can not be empty. Maximum of 400 characters and 50 words in the query.',
    required: true,
});
optional_parameters.push({
    displayName: 'Country',
    name: 'country',
    type: 'options',
    default: 'US',
    description: 'An associated country code. This is used as a hint. Default is "US".',
    options: data_1.SpellcheckCountryCodes.map(({ country, code }) => {
        return {
            name: `${country} (${code})`,
            value: code,
        };
    }),
}, {
    displayName: 'Language',
    name: 'lang',
    type: 'options',
    default: 'en',
    description: 'An associated 2-or-more character language code. Default is "en".',
    options: data_1.SpellcheckLanguageCodes.map(({ language, code }) => {
        return {
            name: `${language} (${code})`,
            value: code,
        };
    }),
});
if (optional_parameters.length > 0) {
    parameters.push({
        displayName: 'Additional Parameters',
        name: 'additionalParameters',
        type: 'collection',
        default: {},
        placeholder: 'Add Parameter',
        options: optional_parameters,
    });
}
exports.default = parameters;
