// @brave/n8n-nodes-brave-search/dist/nodes/BraveSearch/operations/news/parameters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const parameters = [];
const optional_parameters = [];
parameters.push({
    displayName: 'Query',
    name: 'query',
    type: 'string',
    default: '',
    description: 'The user’s search query term. Query can not be empty. Maximum of 400 characters and 50 words in the query.',
    required: true,
}, {
    displayName: 'Count',
    name: 'count',
    type: 'number',
    default: 20,
    description: 'The number of news results returned in response. The maximum is 50. The actual number delivered may be less than requested. Combine this parameter with offset to paginate search results.',
    typeOptions: {
        minValue: 1,
        maxValue: 50,
    },
});
if (optional_parameters.length > 0) {
    parameters.push({
        displayName: 'Additional Parameters',
        name: 'additionalParameters',
        type: 'collection',
        default: {},
        placeholder: 'Add Parameter',
        options: optional_parameters,
    });
}
exports.default = parameters;
