// @brave/n8n-nodes-brave-search/dist/nodes/BraveSearch/operations/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PROPERTIES = exports.OPERATIONS = void 0;
const web_1 = __importDefault(require("./web"));
const news_1 = __importDefault(require("./news"));
const images_1 = __importDefault(require("./images"));
const videos_1 = __importDefault(require("./videos"));
const spellcheck_1 = __importDefault(require("./spellcheck"));
const suggest_1 = __importDefault(require("./suggest"));
const operations = [web_1.default, news_1.default, images_1.default, videos_1.default, spellcheck_1.default, suggest_1.default];
const map = Object.fromEntries(operations.map((e) => [e.key, e]));
exports.OPERATIONS = map;
const all_operations = operations.map(({ details }) => details);
const all_operation_parameters = operations.flatMap(({ parameters }) => parameters);
const properties = [
    {
        displayName: 'Operation',
        description: 'The operation to use for the Brave Search API',
        name: 'operation',
        type: 'options',
        options: all_operations,
        default: 'web',
        required: true,
        noDataExpression: true,
    },
    ...all_operation_parameters,
];
exports.PROPERTIES = properties;
